appbuilder/services/livesync/livesync-service.ts

```diff
@@ -0,0 +1,102 @@
+///<reference path="../../../.d.ts"/>
+"use strict";
+
+import { exportedPromise } from "../../../decorators";
+
+export class ProtonLiveSyncService implements IProtonLiveSyncService {
+	private excludedProjectDirsAndFiles = ["app_resources", "plugins", ".*.tmp", ".ab"];
+
+	private get $liveSyncServiceBase(): ILiveSyncServiceBase {
+		return this.$injector.resolve("liveSyncServiceBase");
+	}
+
+	constructor(private $devicesService: Mobile.IDevicesService,
+		private $errors: IErrors,
+		private $mobileHelper: Mobile.IMobileHelper,
+		private $options: ICommonOptions,
+		private $fs: IFileSystem,
+		private $injector: IInjector,
+		private $project: any,
+		private $logger: ILogger) { }
+
+	@exportedPromise("liveSyncService")
+	public livesync(deviceIdentifiers: IDeviceLiveSyncInfo[], projectDir: string, filePaths?: string[]): IFuture<void> {
+		return (() => {
+			this.$project.projectDir = projectDir;
+			let deviceInfos: any = _.map(deviceIdentifiers, d => {
```

Comment:
I do not want to do this, as the interface is part of the Public API. In case I add it, anyone reading the documentation or the code, will think that this property could be passed as part of the deviceIdentifiers objects.
Of course I could extract separate interface, but I really do not want to do it, as `_.extend` modifies the original object as well.